pages/_middleware.tsx

```diff
@@ -0,0 +1,22 @@
+import { NextFetchEvent, NextRequest, NextResponse } from 'next/server'
+import rateLimit from '../utils/RateLimitUtils'
+import requestIp from 'request-ip'
+
+const limiter = rateLimit({
+    interval: 60 * 1000, // 60 seconds
+    uniqueTokenPerInterval: 500 // Max 500 users per second
+})
+
+export async function middleware(req: NextRequest, ev: NextFetchEvent) {
+    try {
+        const clientIp = requestIp.getClientIp(req)
```

Comment:
Aren't you reading the cloudflare with this?
There is a header that cloudflare uses to forward the real client IP address

Reply:
The package should handle that, shouldnt it? https://www.npmjs.com/package/request-ip
![grafik](https://user-images.githubusercontent.com/58751503/212587784-6d52e95d-712d-4515-a50d-919eb748b725.png)